### PR TITLE
Tomer Heber's coding challenge: async stream reader

### PR DESCRIPTION
**Pull request by iras-auth**
I ran it and tested okay.

### CODE REVIEW

File: AsyncStreamReaderLoggingExample.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import asyncio
+import argparse
+import http.client
+import json
+import logging
+import time
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+
```
Comment (far-authentise) [inline AsyncStreamReaderLoggingExample.py:12]: our default format is 

```
if __name__ == '__main__': 
    args  = parse_args() 
    main( args.username, args.password, args.site) 
```

Please make that small rebuild /remodel of how this is launched/run.

File: AsyncStreamReaderLoggingExample.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import httpx
+
+parser = argparse.ArgumentParser(description="Extracts Authentise events from the stream in an async manner.")
+parser.add_argument("--username", help="your authentise username", required=True)
+parser.add_argument("--password", help="your authentise password", required=True)
+args = parser.parse_args()
```
Comment (far-authentise) [inline AsyncStreamReaderLoggingExample.py]: please add (--base_url) for `dev-auth2.com` or `authentise.pro` etc . This is for the data going into {host} down below.

**Review by far-authentise — CHANGES_REQUESTED** [review]
2 small changes requested.

File: AsyncStreamReaderLoggingExample.py (hunk 1 above)
Comment (TomerHeber) [reply AsyncStreamReaderLoggingExample.py:12]: refactored as requested.

**Review by far-authentise — APPROVED** [review]
👍

File: AsyncStreamReaderLoggingExample.py (hunk 2 above)
Comment (TomerHeber) [reply AsyncStreamReaderLoggingExample.py]: added. If not configured will be set by default to dev-auth2.com

**Pull request merged**